EmbeddedScripts.CSharp/Compilation/CompiledCodeRunner.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Threading.Tasks;
+using EmbeddedScripts.CSharp.Common;
+using EmbeddedScripts.Shared;
+
+namespace EmbeddedScripts.CSharp.Compilation
+{
+    public class CompiledCodeRunner : ICodeRunner
+    {
+        public CompiledCodeRunner(string code, Container container)
+        {
+            Container = container;
+
+            var usingLines = ResolvingCodeGenerator.GenerateUsingLines(Container);
+            var resolveLines = ResolvingCodeGenerator.GenerateResolveLines(Container);
+
+            var codeForCompilation = $@"
+using {typeof(Container).Namespace};
+{usingLines}
+
+public class {ClassName}
+{{
+    public void {MethodName}({nameof(Container)} Container) 
+    {{ 
+       {resolveLines}
+       {code}
+    }} 
+}}";
+
+            var compilation = new CodeCompiler(codeForCompilation, Container).Compilation;
+            Instance = new InstanceCreator(compilation).CreateInstanceOf(ClassName);
```

Comment:
Inconsistent level of abstractions: source code generation, compilation and work with `Instance `abstractions.